docs: add "How to use Vite for the backend" entry to faq

robkorv opened the pull request:
### Description of change

This pull-request resolves #11284

### Pull-Request Checklist
- [x] Code is up-to-date with the `master` branch
- [ ] `npm run format` to apply prettier formatting N/A
- [ ] `npm run test` passes with this change N/A
- [x] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change N/A
- [x] Documentation has been updated to reflect this change
- [x] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/typeorm/typeorm/blob/master/CONTRIBUTING.md)

mguida22 commented on docs/faq.md:
```diff
@@ -266,6 +267,93 @@ module.exports = {
+
+### Option 1: Disable minify
+
+This is the most crude option and will result in significant larger files. Add `build.minify = false` to your config.
```
```suggestion
This is the most crude option and will result in significantly larger files. Add `build.minify = false` to your config.
```

mguida22 commented on docs/faq.md:
```diff
@@ -266,6 +267,93 @@ module.exports = {
+
+### Option 2: Disable esbuild minify identifiers
+
+Vite uses esbuild as default minifier. You can disable mangling of identifiers by adding `esbuild.minifyIdentifiers = false` to your config.
+This will result in smaller file sizes, depending on your code base you will get diminishing returns as all identifiers will be kept at full lengt.
```
```suggestion
Vite uses esbuild as the default minifier. You can disable mangling of identifiers by adding `esbuild.minifyIdentifiers = false` to your config.
This will result in smaller file sizes, but depending on your code base you will get diminishing returns as all identifiers will be kept at full length.
```

mguida22 commented on docs/faq.md:
```diff
@@ -266,6 +267,93 @@ module.exports = {
+
+Using TypeORM in a Vite project is pretty straight forward. However, when you use migrations, you will run into "...migration name is wrong. Migration class name should have a
+JavaScript timestamp appended." errors when running the production build.
+On production builds, files are optimized by default which includes mangling your code in order to minimize file sizes.
```
```suggestion
On production builds, files are [optimized by default](https://vite.dev/config/build-options#build-minify) which includes mangling your code in order to minimize file sizes.
```

mguida22 reviewed (commented):
Thanks @robkorv this is great! I left a few minor comments below.

robkorv commented:
> Hmm. The commits I added via the github gui are not how the should have been. I'll chance them on my machine and force push them.

I'll let you decide. I expected two commits with the message
- docs: apply suggested grammar and spelling changes
- docs: add link to vite documentation

But "Apply suggestions from code review" was also added.... That might be just how you want it.

OSA413 commented:
> I'll let you decide

I think this is ok, we will probably just squash the commits when merging

mguida22 reviewed (approved):
Thank you!
